src/Clang/LowLevel/Core.hs

```diff
@@ -1670,6 +1681,107 @@ clang_Type_getValueType typ = liftIO $ ensureValidType $
+  wrap_EvalResult_getAsStr :: CXEvalResult -> IO CString
+
+foreign import capi unsafe "clang_wrappers.h clang_EvalResult_dispose"
+  nowrapper_EvalResult_dispose :: CXEvalResult -> IO ()
+
+-- | If cursor is a statement declaration tries to evaluate the statement and if
+-- its variable, tries to evaluate its initializer, into its corresponding type.
+--
+-- If it's an expression, tries to evaluate the expression.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#ga6be809ca82538f4a610d9a5b18a10ccb>
+clang_Cursor_Evaluate :: MonadIO m => CXCursor -> m CXEvalResult
+clang_Cursor_Evaluate cursor =
+    liftIO $ onHaskellHeap cursor wrap_Cursor_Evaluate
+
+-- | Returns the kind of the evaluated result.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#gaea912a0620a9c16c1e46fdedf4825955>
+clang_EvalResult_getKind ::
+     MonadIO m
+  => CXEvalResult
+  -> m (SimpleEnum CXEvalResultKind)
+clang_EvalResult_getKind = liftIO . nowrapper_EvalResult_getKind
+
+-- | Returns the evaluation result as integer if the kind is @Int@.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#ga8abe0404897d93813d98bd07a198caa1>
+clang_EvalResult_getAsInt :: MonadIO m => CXEvalResult -> m CInt
+clang_EvalResult_getAsInt = liftIO . nowrapper_EvalResult_getAsInt
+
+-- | Returns the evaluation result as a @long long integer@ if the kind is
+-- @Int@.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#ga488b6b6a445be15e80ffe4816b2086c8>
+clang_EvalResult_getAsLongLong :: MonadIO m => CXEvalResult -> m CLLong
+clang_EvalResult_getAsLongLong = liftIO . nowrapper_EvalResult_getAsLongLong
+
+-- | Returns a non-zero value if the kind is @Int@ and the evaluation result
+-- resulted in an unsigned integer.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#gad72ab38051388e5ed607ce5ce890b2ac>
+clang_EvalResult_isUnsignedInt :: MonadIO m => CXEvalResult -> m CUInt
+clang_EvalResult_isUnsignedInt = liftIO . nowrapper_EvalResult_isUnsignedInt
+
+-- | Returns the evaluation result as an unsigned integer if the kind is @Int@
+-- and @clang_EvalResult_isUnsignedInt@ is non-zero.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#ga448569d83b25514da4a15e6623d4bf4e>
+clang_EvalResult_getAsUnsigned :: MonadIO m => CXEvalResult -> m CULLong
+clang_EvalResult_getAsUnsigned = liftIO . nowrapper_EvalResult_getAsUnsigned
+
+-- | Returns the evaluation result as @double@ if the kind is @double@.
+--
+-- <https://clang.llvm.org/doxygen/group__CINDEX__MISC.html#ga6d140616208f61e24e18abf806aa68a7>
+clang_EvalResult_getAsDouble :: MonadIO m => CXEvalResult -> m CDouble
+clang_EvalResult_getAsDouble = liftIO . nowrapper_EvalResult_getAsDouble
+
+-- | Returns the evaluation result as a constant string if the kind is other
+-- than @Int@ or @float@.
+--
+-- User must not free this pointer, instead call @clang_EvalResult_dispose@ on
+-- the @CXEvalResult@ returned by @clang_Cursor_Evaluate@.
```

Comment:
This comment should be deleted I think, since we `peek` the `String`, so there is nothing that this comment refers to.